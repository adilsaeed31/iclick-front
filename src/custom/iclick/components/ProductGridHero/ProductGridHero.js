import React, { Component } from "react";
import PropTypes from "prop-types";

export default class ProductGridHero extends Component {
  static propTypes = {
    tag: PropTypes.object
  }

  static defaultProps = {
    tag: {}
  }

  render() {
    const {
      tag: { heroMediaUrl }
    } = this.props;

    if (!heroMediaUrl) return null;

    return (
      <section>
        <div>
          <div>
            <img src={heroMediaUrl} alt="Product category" />
          </div>
        </div>
      </section>
    );
  }
}
