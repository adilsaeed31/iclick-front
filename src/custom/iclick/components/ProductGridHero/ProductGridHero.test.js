import React from "react";
import renderer from "react-test-renderer";
import { MuiThemeProvider } from "@material-ui/core/styles";
import theme from "custom/reactionTheme";
import ProductGridHero from "./ProductGridHero";

const tagWithMedia = {
  name: "TagWithMedia",
  _id: "rubhpcpSzrgZuBCEM",
  slug: "tag-with-media",
  isTopLevel: true,
  updatedAt: "2018-05-25 17:48:29.745Z",
  createdAt: "2018-05-25 17:48:29.745Z",
  isDeleted: false,
  isVisible: true,
  groups: [],
  shopId: "J8Bhq3uTtdgwZx3rz",
  heroMediaUrl: "https://www.reactioncommerce.com/assets/careers-images-2.jpg"
};

const tagWithoutMedia = {
  name: "TagWithoutMedia",
  _id: "rubhpcpSzrgZuBCEM",
  slug: "tag-withpoiut-media",
  isTopLevel: true,
  updatedAt: "2018-05-25 17:48:29.745Z",
  createdAt: "2018-05-25 17:48:29.745Z",
  isDeleted: false,
  isVisible: true,
  groups: [],
  shopId: "J8Bhq3uTtdgwZx3rz"
};

test("snapshot of tagWithMedia", () => {
  const component = renderer.create(<MuiThemeProvider theme={theme}>
    <ProductGridHero tag={tagWithMedia} />
  </MuiThemeProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

test("snapshop of tagWithoutMedia", () => {
  const component = renderer.create(<MuiThemeProvider theme={theme}>
    <ProductGridHero tag={tagWithoutMedia} />
  </MuiThemeProvider>);
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
