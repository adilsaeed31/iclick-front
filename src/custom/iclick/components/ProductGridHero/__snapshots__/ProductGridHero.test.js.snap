// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshop of tagWithoutMedia 1`] = `null`;

exports[`snapshot of tagWithMedia 1`] = `
<section
  className="SkProductGridHero-heroGridContainer-2"
>
  <div
    className="MuiGrid-container-3 MuiGrid-spacing-xs-24-27"
  >
    <div
      className="MuiGrid-item-4 MuiGrid-grid-xs-12-43"
    >
      <img
        alt="Product category"
        className="SkProductGridHero-heroImg-1"
        src="https://www.reactioncommerce.com/assets/careers-images-2.jpg"
      />
    </div>
  </div>
</section>
`;
